python/hsfs/feature_group.py

```diff
@@ -216,6 +216,68 @@ def get_tags(self):
         """
         return self._feature_group_engine.get_tags(self)
 
+    def get_parent_feature_groups(self):
+        """Get the parent feature groups that were used to generate this feature
+        group. This is based on explicit provenance.
+        """
+        links_dto = self._feature_group_engine.get_explicit_provenance_links(self)
+        parent_links = []
+        for link in links_dto.upstream:
+            parent_links.append(
+                {
+                    "project": link["project"],
+                    "name": link["name"],
+                    "version": link["version"],
```

Comment:
IMHO this is not a user friendly API. 

- I would return an object and not a dictionary. If you return an object here, the documentation will be able to pick it up and users will be able to know what this method returns, without having to go read the source code.

- I would not call the field project, I would call it `feature_store` to be compatible with the `get_feature_store()` call. and make the API consistent. The project is only using during the connection setup, everywhere else we use the `feature_store` reference.

- If the feature group exists and it's accessible, I would make the call invoke the `get_feature_group` method (from the correct feature store) and store the feature group reference in the object. This will make the APIs more user friendly and consistent with the parent configuration when you create a feature group. When you create a feature group you provide a metadata object, not name/version. So IMHO the method should return the metadata object as well (if possible).